go/service/log_queue.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package service
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+type logQueue struct {
+	active      bool
+	name        string
+	level       keybase1.LogLevel
+	ui          *LogUI
+	handleID    int
+	buffer      chan *logEntry
+	drop        chan bool
+	processDone chan bool
+}
+
+func newLogQueue() *logQueue {
+	// empty until Setup is called
+	return &logQueue{}
+}
+
+func (q *logQueue) Setup(name string, level keybase1.LogLevel, ui *LogUI) error {
+	if q.active {
+		return fmt.Errorf("Setup called on active queue, name = %s, level = %d", name, level)
+	}
+	q.name = name
+	q.level = level
+	q.ui = ui
+	q.buffer = make(chan *logEntry, 10000)
+	q.drop = make(chan bool, 1)
+	q.processDone = make(chan bool, 1)
+
+	q.active = true
+	go q.processQueue()
+
+	return nil
+}
+
+func (q *logQueue) SetHandleID(id int) {
+	q.handleID = id
+}
+
+func (q *logQueue) HandleID() int {
+	return q.handleID
+}
+
+func (q *logQueue) Log(e *logEntry) {
+	if !q.active {
+		return
```

Comment:
It made it slightly easier to have a log queue for every connection so the log handler rpc service. I don't think a bool comparison for n windows is that bad all things considered. But I will look into alternatives.